src/main/java/Duke.java

```diff
@@ -1,10 +1,259 @@
+            String response = parseCommand(input);
+            printMessage(response);
+        }
+    }
+
+    /**
+     * Prints the specified message, followed by a horizontal separator on the next line.
+     * @param message the message to be printed
+     */
+    private static void printMessage(String message) {
+        System.out.println(message);
+        System.out.println(HORIZONTAL_SEPARATOR);
+    }
+
+    /**
+     * Reads user input and separates the command being invoked from its arguments.
+     * @return a String array of the form {command, args}
+     */
+    private static String[] getUserInput() {
+        System.out.print(INPUT_PROMPT);
+        String userInput = in.nextLine();
+        System.out.println(HORIZONTAL_SEPARATOR);
+        String[] userInputTokenized = userInput.trim().split(" ", 2);
+        if (userInputTokenized.length == 2) {
+            return userInputTokenized;
+        }
+        return new String[] {userInputTokenized[0], ""};
+    }
+
+    /**
+     * Generates an error message for an invalid command.
+     * @param usageExample string showing the correct command format
+     * @return generated error message
+     */
+    private static String invalidCommandError(String command, String usageExample) {
+        return String.format(MESSAGE_INCORRECT_COMMAND_FORMAT, command, usageExample);
+    }
+
+    /**
+     * Attempts to add the specified task to the task list.
+     * @param t task to be added
+     * @param successMessage message to be displayed on success
+     */
+    private static String addTask(Task t, String successMessage) {
+        if (numSavedTasks < MAX_TASKS) {
+            savedTasks[numSavedTasks] = t;
+            numSavedTasks++;
+            return String.format(successMessage, t);
+        }
+        return MESSAGE_TOO_MANY_TASKS;
+    }
+
+    /**
+     * Invokes the correct command function based on the command string inputted by the user.
+     */
+    private static String parseCommand(String[] input) {
+        final String command = input[0];
+        final String args = input[1];
+        switch (command) {
+        case COMMAND_BYE:
+            return byeCommand(args);
+        case COMMAND_LIST:
+            return listCommand(args);
+        case COMMAND_TODO:
+            return todoCommand(args);
+        case COMMAND_DEADLINE:
+            return deadlineCommand(args);
+        case COMMAND_EVENT:
+            return eventCommand(args);
+        case COMMAND_MARK:
+            return markCommand(args);
+        case COMMAND_UNMARK:
+            return unmarkCommand(args);
+        default:
+            return MESSAGE_UNKNOWN_COMMAND;
+        }
+    }
+
+    /**
+     * Request to exit the program.
+     */
+    private static String byeCommand(String args) {
+        if (args.equals("")) {
+            isExitRequested = true;
+            return MESSAGE_GOODBYE;
+        }
+        return invalidCommandError(COMMAND_BYE, USAGE_BYE);
```

Comment:
Nice use of invalidCommandError helper function - although this helper function is very short, the name "invalidCommandError" makes it very clear what exactly the code is doing (as opposed to not using a helper function and simply writing the body of the invalidCommandError function here)